### PR TITLE
Export types needed for Egress usage with 2.0

### PR DESCRIPTION
**Pull request by davidzhao**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "compilerOptions": {
     "target": "ES2017" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */,
-    "module": "ES2020" /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', 'es2020', or 'ESNext'. */,
+    "module": "ESNext" /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', 'es2020', or 'ESNext'. */,
```
Comment (lukasIO) [inline tsconfig.json:4]: Are we sure we want to update to `ESNext`?

File: package.json [hunk 2]
```diff
@@ -34,6 +34,7 @@
     "@livekit/changesets-changelog-github": "^0.0.4",
     "@types/node": "^20.10.1",
     "@typescript-eslint/eslint-plugin": "^4.29.1",
+    "@types/node-jose": "^1.1.13",
```
Comment (lukasIO) [inline package.json:37]: `jose` ships already with types ootb. Adding another types package explicitly might just lead to some inaccuracies there. What's the intention behind adding this? (looks like the types are actually for cisco's fork of the library)

**Review by lukasIO — APPROVED** [review]
Exporting the types/classes makes sense.
Don't think we'd want the jose types package and the tsconfig module change.

File: tsconfig.json (hunk 1 above)
Comment (davidzhao) [reply tsconfig.json:4]: it wasn't letting me use await. I can revert if there are other downsides here.

File: package.json (hunk 2 above)
Comment (davidzhao) [reply package.json:37]: ah I'll revert this. the types didn't show up for me in VS code.

**Pull request merged**